lib/resolvers/error.js

```diff
@@ -31,14 +32,39 @@ const _cdsToGraphQLError = (context, err) => {
 }
 
 const _ensureError = error => (error instanceof Error ? error : new Error(error))
+const _clone = obj => Object.create(Object.getPrototypeOf(obj), Object.getOwnPropertyDescriptors(obj))
 
 const handleCDSError = (context, error) => {
   error = _ensureError(error)
-  // TODO: choose log level depending on type of error analogous to OData adapter
-  if (LOG_CDS._error) LOG_CDS.error(error)
+  _log(error)
   return _cdsToGraphQLError(context, error)
 }
 
+const _log = error => {
+  // do not log standard errors, e.g. TypeError
+  if (error.code === undefined && error.details === undefined) return
+
+  // log errors and warnings only
+  if (LOG_CDS.level <= cds.log.levels.WARN) return
+
+  // Clone of the original error object to prevent mutation and unintended side-effects.
+  // Notice that the cloned error is logged to standard output in its default language,
+  // whereas the original error message is locale-dependent as it is usually sent in the
+  // HTTP response to HTTP Clients to be displayed in the user interface.
+  const error2log = _clone(error)
+  if (error.details) error2log.details = error.details.map(error => _clone(error))
+
+  const statusCode = normalizeError(error2log, { locale: i18n.default_language }, error => error).statusCode
```

Comment:
If we do something like this we can avoid relying on `normalizeError` to continue to modify the error in-place:
```suggestion
  const errorClone = _clone(error)
  if (error.details) errorClone.details = error.details.map(error => _clone(error))

  const { statusCode, error: error2log } = normalizeError(errorClone, { locale: i18n.default_language }, error => error)
```

Reply:
I usually prefer this approach, as it is generally beneficial for code readability, allowing the reader to better understand the flow of changes in the error object. However, it might also be confusing in this case, as `normalizeError` mutates the error anyway.

Reply:
Hmmm, idk. All other usages of `normalizeError` in our code base continue to use the error returned by the function call, rather than the error that is passed to it. In theory they should be the same, but that would change as soon as the internals of `normalizeError` change. I would prefer to be consistent and do the same thing here as in the runtime so that changes to the function have consistent results.

I don't necessarily think that it is confusing. The explicit return value makes it clear that it is a result of being processed by `normalizeError`.

Reply:
Ok. It is the same object that is my point.

Reply:
https://github.com/cap-js/graphql/pull/131/commits/66b84e15b3722440cabba8021310f9c8423fff05

Reply:
> Ok. It is the same object that is my point.

You're absolutely right currently. But this way, if that changes in a future `@sap/cds` release, it won't be a breaking change.

Thanks for applying the change :)